src/v2/api/user/UserService.ts

```diff
@@ -76,6 +79,18 @@ export class UserService {
     await this.superheroRepository.updateSuperhero(userId, data);
   }
 
+  async requestNewGroup(id: string, {groupId, isCaptain}: GroupRequestDTO) {
+    await this.studentRepository.update(id, {state: State.PENDING});
```

Comment:
before manipulation with state, check whether this user has approved group, if so deny his request. user can change group only if state is denied or pending